reference_implementations/aws/offline/04_lambda.tf

```diff
@@ -0,0 +1,151 @@
+resource "aws_iam_role" "lambda_role" {
+  name = "BertParaphraseModelLambdaRoleTF"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+# aws_iam_role_policy: For creating inline, role-specific policies.
+resource "aws_iam_role_policy" "lambda_logs_policy" {
+  name = "lambda_role_logs_policy"
+  role = aws_iam_role.lambda_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        "Action": [
+            "logs:CreateLogGroup",
+            "logs:CreateLogStream",
+            "logs:DescribeLogGroups",
+            "logs:DescribeLogStreams",
+            "logs:PutLogEvents",
+            "logs:GetLogEvents",
+            "logs:FilterLogEvents"
+        ],
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "lambda_sagemaker_policy" {
+  name = "lambda_role_sagemaker_policy"
+  role = aws_iam_role.lambda_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = "sagemaker:InvokeEndpoint"
+        Resource = "arn:aws:sagemaker:${var.region}:${local.aws_account_id}:endpoint/*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "lambda_sagemaker_featurestore_policy" {
+  name = "lambda_role_sagemaker_featurestore_policy"
+  role = aws_iam_role.lambda_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = [
+          "sagemaker:GetRecord",
+          "sagemaker:PutRecord",
+          "sagemaker:ListFeatureGroups",
+          "sagemaker:BatchGetRecord"
+        ]
+        Resource = "arn:aws:sagemaker:${var.region}:${local.aws_account_id}:feature-group/*"
+      }
+    ]
+  })
+}
+
+# Attach AmazonRedshiftDataFullAccess policy to the role
+# aws_iam_policy_attachment: For attaching existing managed policies 
+# (either AWS-managed or your own custom policies) to roles, users, or groups.
+resource "aws_iam_policy_attachment" "redshift_data_access" {
+  name       = "lambda_role_redshift_data_access_attachment"
+  roles      = [aws_iam_role.lambda_role.id]
+  policy_arn = "arn:aws:iam::aws:policy/AmazonRedshiftFullAccess"
+}
+
+# resource "aws_iam_policy_attachment" "sqs_access" {
+#   name       = "lambda_role_redshift_data_access_attachment"
+#   roles      = [aws_iam_role.lambda_role.id]
+#   policy_arn = "arn:aws:iam::aws:policy/AWSLambdaSQSQueueExecutionRole"
+# }
```

Comment:
If this is not needed we should remove instead of leaving it commented. If we need it later we can find it on github commit history.